genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaClusterServiceImpl.java

```diff
@@ -318,6 +318,68 @@ public void removeAllConfigsForCluster(
      * {@inheritDoc}
      */
     @Override
+    public void addDependenciesForCluster(
```

Comment:
Commands don't have dependencies as of now (although there are hints around the code that seem to indicate this was the case previously, or maybe it's a partially implemented feature that was never fully realized).
For example `createEntityInstanceDependenciesDirectory` knows how to create a folder for 'command' dependencies, but it's never used for that purpose.

If we add dependencies to commands (now in the list of things to do) we can indeed push some this and more up to some common parent and drop a bunch of duplicate code in JPA/DTO/Service/REST/...